### PR TITLE
refactor: replace yargs-parser with Node.js util.parseArgs

### PR DESCRIPTION
**Pull request by ilteoood**
## Summary
- Replace `yargs-parser` dependency with Node.js built-in `util.parseArgs`
- Reduces external dependencies

## Test plan
- [x] All 58 tests pass
- [x] Lint passes

### CODE REVIEW

**Review by Tony133 — APPROVED** [review]

File: lib/args.js [hunk 1]
```diff
@@ -1,35 +1,76 @@
...
-    'silent',
-    'gh-release-edit',
-    'gh-release-draft',
-    'gh-release-prerelease',
-    'gh-release-body'
-  ]
+const booleanOptions = new Set([
+  'help', 'major', 'dry-run', 'no-verify', 'silent',
+  'gh-release-edit', 'gh-release-draft', 'gh-release-prerelease', 'gh-release-body'
+])
```
Comment (Eomm) [inline lib/args.js]: this repeats the `module.exports.argsNames`

File: lib/args.js [hunk 2]
```diff
@@ -53,61 +94,75 @@ module.exports = function parsedArgs (args) {
...
+    npmOtp: values['npm-otp'],
+    tag: values.tag,
+    verbose: values.verbose,
+    semver: values.semver,
+    major: values.major
+  }
 }
 
-module.exports.argsNames = argsNames
+module.exports.argsNames = {
```
Comment (Eomm) [inline lib/args.js]: can't we generate it by looping `optionsConfig`?

**Review by Tony133 — APPROVED** [review]

File: lib/args.js [hunk 3]
```diff
@@ -1,35 +1,71 @@
...
+  remote: { type: 'string', short: 'r' },
+  branch: { type: 'string', short: 'b' },
+  'from-commit': { type: 'string' },
+  'to-commit': { type: 'string' },
+  'npm-access': { type: 'string', short: 'a' },
+  'npm-dist-tag': { type: 'string' },
+  'npm-otp': { type: 'string' },
+  'gh-token': { type: 'string', short: 'k' },
+  arg: { type: 'string' },
+  'gh-group-by-label': { type: 'string', short: 'l', multiple: true }
```
Comment (Eomm) [inline lib/args.js:29]: The `normalizeArgs` preprocessing could be removed entirely. Since this is a breaking change and you are the primary users, consider dropping support for the edge cases `normalizeArgs` handles. `parseArgs` accepts a clean format natively — if that was tested, the whole function is unnecessary overhead.

File: test/args.test.js [hunk 4]
```diff
@@ -10,27 +10,27 @@ test('parse all args', t => {
 
   const argv = [
     '--arg', 'arg',
-    '--help', 'true',
+    '--help=true',
```
Comment (Eomm) [inline test/args.test.js]:
```suggestion
    '--help',
```

File: test/args.test.js [hunk 5]
```diff
@@ -10,27 +10,27 @@ test('parse all args', t => {
...
-    '--major', 'true',
-    '--dry-run', 'true',
+    '--major=true',
+    '--dry-run=true',
     '--remote', 'upstream',
     '--branch', 'v1',
     '--from-commit', 'commitFrom',
     '--to-commit', 'commitTo',
-    '--no-verify', 'true',
+    '--no-verify=true',
```
Comment (Eomm) [inline test/args.test.js]:
```suggestion
    '--no-verify',
```

File: test/args.test.js [hunk 6]
```diff
@@ -10,27 +10,27 @@ test('parse all args', t => {
...
     '--npm-dist-tag', 'next',
     '--gh-token', 'MY_KEY',
-    '--gh-release-edit', 'true',
-    '--gh-release-draft', 'true',
-    '--gh-release-prerelease', 'true',
-    '--gh-release-body', 'true',
+    '--gh-release-edit=true',
+    '--gh-release-draft=true',
+    '--gh-release-prerelease=true',
+    '--gh-release-body=true',
```
Comment (Eomm) [inline test/args.test.js]:
```suggestion
    '--gh-release-body',
```

**Comment by Tony133** [comment]
@Eomm can we do the merge?

**Review by Eomm — APPROVED** [review]
